src/hooks/HMSRoomProvider.tsx

```diff
@@ -75,14 +76,41 @@ export const HMSRoomProvider: React.FC<HMSRoomProviderProps> = ({
   );
 };
 
-/*
-UseHMSStore is a read only hook which can be passed a selector to read data.
-The hook can only be used in a component if HMSRoomProvider is present in its ancestors.
+/**
+ * `useHMSStore` is a read only hook which can be passed a selector to read data.
+ * The hook can only be used in a component if HMSRoomProvider is present in its ancestors.
  */
 export const useHMSStore = makeHMSStoreHook(HMSContext);
 
+/**
+ * `useHMSVanillaStore` is a read only hook which returns the vanilla HMSStore.
+ * Usage:
+ * ```
+ * const hmsStore = useHMSVanillaStore();
+ * const dominantSpeaker = hmsStore.getState(selectDominantSpeaker);
+ * ```
+ *
+ * Note: Refrain from using this in React components.
```

Comment:
it's a hook, so it has to be used in a react component right?
we can note here that the component won't be re rendered if state is fetched this way. 

Eventually I'm planning for a performance page on docs where this can be recommended for certain use cases.

Reply:
I'll rephrase it. The way this could be used is getting the vanilla store in a parent component and passing this as param to another function. 

What I meant was to be cautious when using this. In almost every case useHMSStore would work and clients need not use this vanilla store in React components.